File: algorithmProblemSolving/programmers/level1/소수 찾기.js
function solution(n) {
    var answer = [];
    const tmpSet = new Set();
    //tmpSet.add(2);
    for(let i=1;i<=n;i+=2){
        tmpSet.add(i);
    }
    tmpSet.delete(1);
    for(let i=3;i<Math.sqrt(n);i++){
        if(tmpSet.has(i)){
            for(let k=i*2; k<=n; k+=i){    
                tmpSet.delete(k);
            }
        }
    }
    return tmpSet.size+1;
}

File: algorithmProblemSolving/programmers/level1/소수 찾기.cpp
#include <string>
#include <vector>

using namespace std;
vector<bool> nums(1000001,true);
int solution(int n) {
    int answer = 0;
    nums[0] = false;
    nums[1] = false;
    bool check = true;
    for(int i=2;i<n+1;i++){
        if(!nums[i]) continue;
        for(int j=i*2;j<n+1;j=j+i){
            nums[j] = false;
        }
    }
    for(int i=0;i<n+1;i++){
        if(nums[i]) answer++;
    }
    return answer;
}